frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1755,6 +1755,31 @@ const IR::Node* TypeInference::preorder(IR::EntriesList* el) {
     return el;
 }
 
+const IR::Node* TypeInference::preorder(IR::Type_SerEnum* se) {
+    const IR::Type_Bits* type;
+    if (se->type == nullptr) {
```

Comment:
Here you should just lookup `se->type` in the typeMap and check it is one of the allowed types.

Reply:
From the `p4parser.ypp` code, either `se->type` is `nullptr` or  `se->nameType` is `nullptr`.  So this check is checking for `se->type` being `nullptr` so that the `nameType` can be converted to Type_Bits.  I will add a comment above the check to explain.